frontend/src/components/room/chat/VideoChat.tsx

```diff
@@ -0,0 +1,100 @@
+import { memo, useMemo, useState } from "react";
+import {
+  MicOffOutlined,
+  MicOutlined,
+  VideocamOffOutlined,
+  VideocamOutlined,
+  VolumeOffOutlined,
+  VolumeUpOutlined,
+} from "@mui/icons-material";
+import { Button, Grid, IconButton, Stack } from "@mui/material";
+
+import { Center } from "src/components/Center";
+import { useChat } from "src/contexts/ChatContext";
+
+import { LocalVideoChatParticipant } from "./LocalVideoChatParticipant";
+import { RemoteVideoChatParticipant } from "./RemoteVideoChatParticipant";
+
+// eslint-disable-next-line react/display-name
+export const VideoChat = memo(
+  ({ name, partnerName }: { name?: string; partnerName?: string }) => {
+    const {
+      isVideoChatEnabled,
+      enableVideoChat,
+      selfVideoParticipant,
+      videoParticipants,
+      isAudioEnabled,
+      setIsAudioEnabled,
+      isVideoEnabled,
+      setIsVideoEnabled,
+    } = useChat();
+
+    const [isMuted, setIsMuted] = useState<boolean>(false);
+
+    const partner = useMemo(
+      () => Array.from(videoParticipants.values())[0],
+      [videoParticipants],
+    );
+
+    return isVideoChatEnabled ? (
+      <Stack spacing={1} sx={{ height: "100%" }}>
+        <Grid container sx={{ flex: 1 }}>
+          {/* Material UI uses negative margins to achieve spacing and that 
+         affects our video layout so we hack around it. */}
+          <Grid xs={5.75} item>
+            <LocalVideoChatParticipant
+              name={name}
+              participant={selfVideoParticipant}
+            />
+          </Grid>
+          <Grid xs={0.5} />
+          <Grid xs={5.75} item sx={{ position: "relative", flex: 1 }}>
+            <RemoteVideoChatParticipant
+              name={partnerName}
+              participant={partner}
+              isMuted={isMuted}
+            />
+          </Grid>
+        </Grid>
+        <Stack direction="row" spacing={1} justifyContent="space-between">
+          <Stack direction="row" spacing={1} justifyContent="space-between">
+            <IconButton
+              size="small"
+              onClick={() => setIsVideoEnabled(!isVideoEnabled)}
+            >
+              {isVideoEnabled ? (
+                <VideocamOutlined fontSize="small" color="error" />
+              ) : (
+                <VideocamOffOutlined fontSize="small" />
+              )}
+            </IconButton>
+            <IconButton
+              size="small"
+              onClick={() => setIsAudioEnabled(!isAudioEnabled)}
+            >
+              {isAudioEnabled ? (
+                <MicOutlined fontSize="small" color="error" />
+              ) : (
+                <MicOffOutlined fontSize="small" />
+              )}
+            </IconButton>
+          </Stack>
+          <IconButton
+            size="small"
+            onClick={() => setIsMuted((muted) => !muted)}
+          >
+            {isMuted ? (
+              <VolumeOffOutlined fontSize="small" color="warning" />
```

Comment:
Should we make the color scheme of this component to be consistent with the mic and videocam outlines? (ie red on enable, grey on disable)

Alternatively we can use blue on enable, grey/red on disable. But I understand that there is a UI/UX rationale behind using the existing color scheme so we can also stick to the existing color scheme

Reply:
I'll change this to show red when the other user is muted to stay in line with the other colors.